lib/BackgroundJob/CheckHostedSignalingServer.php

```diff
@@ -67,9 +68,17 @@ protected function run($argument): void {
 		$accountId = new AccountId($accountId);
 		try {
 			$accountInfo = $this->hostedSignalingServerService->fetchAccountInfo($accountId);
-		} catch (HostedSignalingServerAPIException $e) { // API or connection issues
-			// do nothing and just try again later
-			return;
+		} catch (HostedSignalingServerAPIException $e) {
+			if ($e->getCode() === Http::STATUS_NOT_FOUND) {
+				// Account was deleted, so remove the information locally
+				$accountInfo = ['status' => 'deleted'];
+			} elseif ($e->getCode() === Http::STATUS_UNAUTHORIZED) {
+				// Account is expired and deletion is pending unless it's reactivated.
```

Comment:
it seems previously expired accounts always received a 200 with body. Now it's a 401. Deleted accounts receive a 404.

So both cases are now specifically catched and handled so the status notifications happen as planned.